From 831167972f67f85ac23326bb696aa1b4c6350895 Mon Sep 17 00:00:00 2001
From: Petr Hodina <petr.hodina@protonmail.com>
Date: Tue, 13 Jan 2026 22:24:47 +0000
Subject: [PATCH] FROMGIT arm64: defconfig: Enable configs for Fairphone 3, 4,
 5 smartphones

Enable base options for MSM8953 and SM6350 SoCs and device-specific
options which are used on the Fairphone 3, Fairphone 4 and Fairphone 5.

* MSM8953 SoC
    MSM_GCC_8953, INTERCONNECT_QCOM_MSM8953
* Fairphone 3
    NFC_NXP_NCI*, TOUCHSCREEN_HIMAX_HX83112B, DRM_PANEL_HIMAX_HX83112B

* SM6350 SoC
    SM_CAMCC_6350, SM_DISPCC_6350, SM_GCC_6350, SM_GPUCC_6350,
    SM_VIDEOCC_6350, INTERCONNECT_QCOM_SM6350
* Fairphone 4
    DRM_PANEL_HIMAX_HX83112A

* Fairphone 5
    DRM_PANEL_RAYDIUM_RM692E5, TYPEC_MUX_PTN36502, INPUT_AW86927

* QCOM PMICs (used on multiple of the devices)
    BACKLIGHT_QCOM_WLED, MFD_QCOM_PM8008, REGULATOR_QCOM_PM8008,
    LEDS_CLASS_FLASH, LEDS_QCOM_FLASH

* USB audio offloading (enabled on Fairphone 4 & 5)
    SND_USB_AUDIO*, SND_SOC_USB, SND_SOC_QDSP6_USB, USB_XHCI_SIDEBAND

Signed-off-by: Luca Weiss <luca.weiss@fairphone.com>
Reviewed-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Reviewed-by: Konrad Dybcio <konrad.dybcio@oss.qualcomm.com>
Link: https://lore.kernel.org/r/20251014-arm64-defconfig-fp345-v1-1-53fe1eeb598d@fairphone.com
Signed-off-by: Bjorn Andersson <andersson@kernel.org>
---
 arch/arm64/configs/fairphone.config | 41 +++++++++++++++++++++++++++++
 1 file changed, 41 insertions(+)
 create mode 100644 arch/arm64/configs/fairphone.config

diff --git a/arch/arm64/configs/fairphone.config b/arch/arm64/configs/fairphone.config
new file mode 100644
index 000000000000..8804bcd21d25
--- /dev/null
+++ b/arch/arm64/configs/fairphone.config
@@ -0,0 +1,41 @@
+# Fairphone config fragment
+# MSM8953 SoC
+CONFIG_MSM_GCC_8953=y
+CONFIG_INTERCONNECT_QCOM_MSM8953=y
+
+# SM6350 SoC
+CONFIG_INTERCONNECT_QCOM_SM6350=y
+CONFIG_SM_CAMCC_6350=m
+CONFIG_SM_DISPCC_6350=m
+CONFIG_SM_GCC_6350=y
+CONFIG_SM_GPUCC_6350=m
+CONFIG_SM_VIDEOCC_6350=m
+
+# Fairphone 3
+CONFIG_NFC_NXP_NCI=m
+CONFIG_NFC_NXP_NCI_I2C=m
+CONFIG_TOUCHSCREEN_HIMAX_HX83112B=m
+CONFIG_DRM_PANEL_HIMAX_HX83112B=m
+
+# Fairphone 4
+CONFIG_DRM_PANEL_HIMAX_HX83112A=m
+
+# Fairphone 5
+CONFIG_INPUT_AW86927=m
+CONFIG_DRM_PANEL_RAYDIUM_RM692E5=m
+CONFIG_TYPEC_MUX_PTN36502=m
+
+# QCOM PMICs
+CONFIG_MFD_QCOM_PM8008=m
+CONFIG_REGULATOR_QCOM_PM8008=m
+CONFIG_BACKLIGHT_QCOM_WLED=m
+
+CONFIG_LEDS_CLASS_FLASH=m
+CONFIG_LEDS_QCOM_FLASH=m
+
+# USB audio offloading (Fairphone 4 & 5)
+CONFIG_SND_USB_AUDIO=m
+CONFIG_SND_USB_AUDIO_QMI=m
+CONFIG_SND_SOC_USB=m
+CONFIG_SND_SOC_QDSP6_USB=m
+CONFIG_USB_XHCI_SIDEBAND=y
-- 

